### ISSUE TITLE
Getting consistent Error  when installing for Nerf Studio

### ISSUE THREAD

**Issue by Lievargus** [NVlabs/tiny-cuda-nn#498]
Every time I try to install tiny-cuda-nn I get the error "[WinError 5] Access is denied". I am running things out of anaconda as an administrator. I have been trouble shooting for 2 days and am out of ideas.

**Below is my current versions:**
> (nerfstudio) C:\Users\Monk2>"C:\Program Files (x86)\Microsoft Visual Studio\2022\BuildTools\VC\Auxiliary\Build\vcvars64.bat"
> **********************************************************************
> ** Visual Studio 2022 Developer Command Prompt v17.13.6
> ** Copyright (c) 2022 Microsoft Corporation
> **********************************************************************
> [vcvarsall.bat] Environment initialized for: 'x64'
> 
> (nerfstudio) C:\Users\Monk2>where cl
> C:\Program Files (x86)\Microsoft Visual Studio\2022\BuildTools\VC\Tools\MSVC\14.43.34808\bin\Hostx64\x64\cl.exe
> 
> (nerfstudio) C:\Users\Monk2>cl
> Microsoft (R) C/C++ Optimizing Compiler Version 19.43.34810 for x64
> Copyright (C) Microsoft Corporation.  All rights reserved.
> 
> usage: cl [ option... ] filename... [ /link linkoption... ]
> 
> (nerfstudio) C:\Users\Monk2>python --version
> Python 3.8.20
> 
> (nerfstudio) C:\Users\Monk2>python -m pip --version
> pip 25.0.1 from C:\ProgramData\anaconda3\envs\nerfstudio\lib\site-packages\pip (python 3.8)
> 
> (nerfstudio) C:\Users\Monk2>python -c "import torch; print(torch.__version__); print(torch.version.cuda); print(torch.cuda.get_device_capability())"
> 2.1.2+cu118
> 11.8
> (8, 9)
> 
> (nerfstudio) C:\Users\Monk2>conda list
> # packages in environment at C:\ProgramData\anaconda3\envs\nerfstudio:
> #
> # Name                    Version                   Build  Channel
> bzip2                     1.0.8                h2bbff1b_6
> ca-certificates           2025.2.25            haa95532_0
> certifi                   2025.1.31                pypi_0    pypi
> charset-normalizer        3.4.1                    pypi_0    pypi
> cmake                     4.0.0                    pypi_0    pypi
> cuda-cccl                 11.8.89                       0    nvidia/label/cuda-11.8.0
> cuda-command-line-tools   11.8.0                        0    nvidia/label/cuda-11.8.0
> cuda-compiler             11.8.0                        0    nvidia/label/cuda-11.8.0
> cuda-cudart               11.8.89                       0    nvidia/label/cuda-11.8.0
> cuda-cudart-dev           11.8.89                       0    nvidia/label/cuda-11.8.0
> cuda-cuobjdump            11.8.86                       0    nvidia/label/cuda-11.8.0
> cuda-cupti                11.8.87                       0    nvidia/label/cuda-11.8.0
> cuda-cuxxfilt             11.8.86                       0    nvidia/label/cuda-11.8.0
> cuda-documentation        11.8.86                       0    nvidia/label/cuda-11.8.0
> cuda-libraries            11.8.0                        0    nvidia/label/cuda-11.8.0
> cuda-libraries-dev        11.8.0                        0    nvidia/label/cuda-11.8.0
> cuda-memcheck             11.8.86                       0    nvidia/label/cuda-11.8.0
> cuda-nsight-compute       11.8.0                        0    nvidia/label/cuda-11.8.0
> cuda-nvcc                 11.8.89                       0    nvidia/label/cuda-11.8.0
> cuda-nvdisasm             11.8.86                       0    nvidia/label/cuda-11.8.0
> cuda-nvml-dev             11.8.86                       0    nvidia/label/cuda-11.8.0
> cuda-nvprof               11.8.87                       0    nvidia/label/cuda-11.8.0
> cuda-nvprune              11.8.86                       0    nvidia/label/cuda-11.8.0
> cuda-nvrtc                11.8.89                       0    nvidia/label/cuda-11.8.0
> cuda-nvrtc-dev            11.8.89                       0    nvidia/label/cuda-11.8.0
> cuda-nvtx                 11.8.86                       0    nvidia/label/cuda-11.8.0
> cuda-nvvp                 11.8.87                       0    nvidia/label/cuda-11.8.0
> cuda-profiler-api         11.8.86                       0    nvidia/label/cuda-11.8.0
> cuda-sanitizer-api        11.8.86                       0    nvidia/label/cuda-11.8.0
> cuda-toolkit              11.8.0                        0    nvidia/label/cuda-11.8.0
> cuda-tools                11.8.0                        0    nvidia/label/cuda-11.8.0
> cuda-visual-tools         11.8.0                        0    nvidia/label/cuda-11.8.0
> filelock                  3.16.1                   pypi_0    pypi
> fsspec                    2025.3.0                 pypi_0    pypi
> idna                      3.10                     pypi_0    pypi
> jinja2                    3.1.6                    pypi_0    pypi
> libcublas                 11.11.3.6                     0    nvidia/label/cuda-11.8.0
> libcublas-dev             11.11.3.6                     0    nvidia/label/cuda-11.8.0
> libcufft                  10.9.0.58                     0    nvidia/label/cuda-11.8.0
> libcufft-dev              10.9.0.58                     0    nvidia/label/cuda-11.8.0
> libcurand                 10.3.0.86                     0    nvidia/label/cuda-11.8.0
> libcurand-dev             10.3.0.86                     0    nvidia/label/cuda-11.8.0
> libcusolver               11.4.1.48                     0    nvidia/label/cuda-11.8.0
> libcusolver-dev           11.4.1.48                     0    nvidia/label/cuda-11.8.0
> libcusparse               11.7.5.86                     0    nvidia/label/cuda-11.8.0
> libcusparse-dev           11.7.5.86                     0    nvidia/label/cuda-11.8.0
> libffi                    3.4.4                hd77b12b_1
> libnpp                    11.8.0.86                     0    nvidia/label/cuda-11.8.0
> libnpp-dev                11.8.0.86                     0    nvidia/label/cuda-11.8.0
> libnvjpeg                 11.9.0.86                     0    nvidia/label/cuda-11.8.0
> libnvjpeg-dev             11.9.0.86                     0    nvidia/label/cuda-11.8.0
> libuv                     1.48.0               h827c3e9_0
> lz4-c                     1.9.4                h2bbff1b_1
> markupsafe                2.1.5                    pypi_0    pypi
> mpmath                    1.3.0                    pypi_0    pypi
> networkx                  3.1                      pypi_0    pypi
> ninja                     1.12.1               hc790b64_0    conda-forge
> nsight-compute            2022.3.0.22                   0    nvidia/label/cuda-11.8.0
> numpy                     1.24.4                   pypi_0    pypi
> openssl                   3.5.0                ha4e3fda_0    conda-forge
> pillow                    10.4.0                   pypi_0    pypi
> pip                       25.0.1                   pypi_0    pypi
> python                    3.8.20               h8205438_0
> requests                  2.32.3                   pypi_0    pypi
> setuptools                75.3.2                   pypi_0    pypi
> sqlite                    3.45.3               h2bbff1b_0
> sympy                     1.13.3                   pypi_0    pypi
> torch                     2.1.2+cu118              pypi_0    pypi
> torchvision               0.16.2+cu118             pypi_0    pypi
> typing-extensions         4.13.2                   pypi_0    pypi
> ucrt                      10.0.22621.0         h57928b3_1    conda-forge
> urllib3                   2.2.3                    pypi_0    pypi
> vc                        14.42                haa95532_5
> vc14_runtime              14.42.34438         hfd919c2_26    conda-forge
> vs2015_runtime            14.42.34438         h7142326_26    conda-forge
> wheel                     0.45.1                   pypi_0    pypi
> xz                        5.6.4                h4754444_1
> zlib                      1.2.13               h8cc25b3_1
> zstd                      1.5.6                h8880b57_0

**Below is the exact output:**

> (nerfstudio) C:\Users\Monk2>pip install git+https://github.com/NVlabs/tiny-cuda-nn/#subdirectory=bindings/torch
> Collecting git+https://github.com/NVlabs/tiny-cuda-nn/#subdirectory=bindings/torch
>   Cloning https://github.com/NVlabs/tiny-cuda-nn/ to c:\users\monk2\appdata\local\temp\pip-req-build-xf5c2it4
>   Running command git clone --filter=blob:none --quiet https://github.com/NVlabs/tiny-cuda-nn/ 'C:\Users\Monk2\AppData\Local\Temp\pip-req-build-xf5c2it4'
>   Resolved https://github.com/NVlabs/tiny-cuda-nn/ to commit 075158a70b87dba8729188a9cadc9411cfa4b71d
>   Running command git submodule update --init --recursive -q
>   Preparing metadata (setup.py) ... done
> Building wheels for collected packages: tinycudann
>   Building wheel for tinycudann (setup.py) ... error
>   error: subprocess-exited-with-error
> 
>   × python setup.py bdist_wheel did not run successfully.
>   │ exit code: 1
>   ╰─> [23 lines of output]
>       C:\Users\Monk2\AppData\Local\Temp\pip-req-build-xf5c2it4\bindings/torch\setup.py:5: DeprecationWarning: pkg_resources is deprecated as an API. See https://setuptools.pypa.io/en/latest/pkg_resources.html
>         from pkg_resources import parse_version
>       Building PyTorch extension for tiny-cuda-nn version 1.7
>       Obtained compute capability 89 from PyTorch
>       Detected CUDA version 11.8
>       Targeting C++ standard 17
>       running bdist_wheel
>       running build
>       running build_py
>       creating build\lib.win-amd64-cpython-38\tinycudann
>       copying tinycudann\modules.py -> build\lib.win-amd64-cpython-38\tinycudann
>       copying tinycudann\__init__.py -> build\lib.win-amd64-cpython-38\tinycudann
>       running egg_info
>       creating tinycudann.egg-info
>       writing tinycudann.egg-info\PKG-INFO
>       writing dependency_links to tinycudann.egg-info\dependency_links.txt
>       writing top-level names to tinycudann.egg-info\top_level.txt
>       writing manifest file 'tinycudann.egg-info\SOURCES.txt'
>       reading manifest file 'tinycudann.egg-info\SOURCES.txt'
>       writing manifest file 'tinycudann.egg-info\SOURCES.txt'
>       copying tinycudann\bindings.cpp -> build\lib.win-amd64-cpython-38\tinycudann
>       running build_ext
>       error: [WinError 5] Access is denied
>       [end of output]
> 
>   note: This error originates from a subprocess, and is likely not a problem with pip.
>   ERROR: Failed building wheel for tinycudann
>   Running setup.py clean for tinycudann
> Failed to build tinycudann
> ERROR: Failed to build installable wheels for some pyproject.toml based projects (tinycudann)

**Comment by nicholaswbowen** [comment]
I am also having this when I follow the guide for nerf-studio.

On windows 11 with mobile 4070.

**Comment by Lievargus** [comment]
@nicholaswbowen 
I ended up having to do a clean install and use a very specific version and edition of visual studio that is not publicly available using normal means. You can find the steps below:
https://github.com/NVlabs/tiny-cuda-nn/issues/494#issuecomment-2815507366
